File: Java Labs/Patterns On Java/src/com/pattern/mediator/PaymentSystem.java
package com.pattern.mediator;

public class PaymentSystem implements Component {
    private Mediator mediator;

    @Override
    public void setMediator(Mediator mediator) {
        this.mediator = mediator;
    }

    @Override
    public void notify(String event) {
        System.out.println("Payment System received notification: " + event);
        // Additional logic related to payment processing...
    }

    public void processPayment(double amount) {
        // Payment processing logic...
        mediator.notify(this, "Payment processed for amount: $" + amount);
    }
}

File: Python Labs/com/pattern/Mediator/PaymentSystem.py
from com.pattern.Mediator.Component import Component


class PaymentSystem(Component):
    def __init__(self):
        self.mediator = None

    def set_mediator(self, mediator):
        self.mediator = mediator

    def notify(self, event):
        print(f"Payment System received notification: {event}")
        # Additional logic related to payment processing...

    def process_payment(self, amount):
        # Payment processing logic...
        self.mediator.notify(self, f"Payment processed for amount: ${amount}")
